identity-service/src/utils/relayUtils.js

```diff
@@ -93,6 +94,51 @@ const getInstructionEnum = (instruction) => {
   return null
 }
 
+/**
+ * Map of instruction name to the enum used in the token program.
+ * Incomplete, but more entries can be found here:
+ * https://github.com/solana-labs/solana-program-library/blob/0a1cfdb90ccfa7745fd2aa35b1ad38ec8bcb6f88/token/program/src/instruction.rs#L523C15-L523C16
+ */
+const tokenInstructionEnum = /** @type {const} */ ({
+  transferChecked: 12
+})
+/**
+ * Map of account name to the index used in the token program's transfer_checked instruction
+ */
+const transferAccountIndices = /** @type {const} */ ({
+  sender: 0,
+  mint: 1,
+  receiver: 2,
+  authority: 3
+})
+
+/**
+ * Checks to see if the instruction is an allowed token transfer.
+ * Currently the only allowed transfers are to USDC userbanks
+ * @param {Instruction} instruction
+ * @param {string} walletAddress
+ * @returns true if the instruction is allowed
+ */
+const isTransferToUserbank = async (instruction, walletAddress) => {
+  if (
+    getInstructionEnum(instruction) === tokenInstructionEnum.transferChecked
+  ) {
+    const mint = instruction.keys[transferAccountIndices.mint].pubkey
+    const receiverAccount =
+      instruction.keys[transferAccountIndices.receiver].pubkey
+    if (mint === usdcMintAddress) {
+      const derivedUserbank = await audiusLibsWrapper
```

Comment:
Note that I went with deriving the userbank rather than hitting discovery, as the userbank might have just been created and not indexed yet